Cython/Compiler/Annotate.py

```diff
@@ -315,7 +315,7 @@ def annotate(match):
     br'(?P<py_macro_api>Py[A-Z][a-z]+_[A-Z][A-Z_]+)|'
     br'(?P<py_c_api>Py[A-Z][a-z]+_[A-Z][a-z][A-Za-z_]*)'
     br')(?=\()|'       # look-ahead to exclude subsequent '(' from replacement
-    br'(?P<error_goto>(?:(?<=;) *if [^;]* +)?__PYX_ERR\([^)]+\))'
+    br'(?P<error_goto>(?:(?<=;) *if \((?:(?!if[ (]).)*\) +)?__PYX_ERR\([^)]+\))'
```

Comment:
Hmm – so, if a nested `if (…)` appears, the first closing bracket belongs to the inner `if`, which is not followed by a ` __PYX_ERR`, so it needs to unread the closing bracket and accept it as `.` (any) character instead, and then continue to parse for another closing bracket. That's at least a bit of backtracking involved.

Atomic groups are new in Py3.11 and not spelled `(?!` but `(?>`.
https://docs.python.org/3/library/re.html#regular-expression-syntax